api/app/src/external/commonwell/link.ts

```diff
@@ -0,0 +1,181 @@
+import { Person, getId, LOLA } from "@metriport/commonwell-sdk";
+
+import { commonWell, metriportQueryMeta, apiUrl, CW_ID_URL_ENCODED_PREFIX } from "./api";
+import { Link, LinkSource } from "../../routes/medical/schemas/link";
+import { Config } from "../../shared/config";
+
+const rootId = Config.getSystemRootOID();
+
+export const linkPatientToCommonwellPerson = async (
+  personId: string,
+  patientNumber: number
+): Promise<string | undefined> => {
+  const referenceLink = createReferenceLink(patientNumber);
+  try {
+    const link = await commonWell.patientLink(metriportQueryMeta, personId, referenceLink);
+    const networkLinks = await commonWell.getPatientsLinks(
+      metriportQueryMeta,
+      `${patientNumber}${CW_ID_URL_ENCODED_PREFIX}${rootId}`
+    );
+
+    if (networkLinks._embedded && networkLinks._embedded.networkLink?.length) {
+      const lola1Links = networkLinks._embedded.networkLink.filter(
+        link => link.assuranceLevel === LOLA.level_1
+      );
+
+      lola1Links.forEach(async link => {
+        if (link._links?.upgrade?.href) {
+          await commonWell.upgradeOrDowngradeNetworkLink(
+            metriportQueryMeta,
+            link._links?.upgrade?.href
+          );
+        }
+      });
+    }
+
+    return link._links?.self?.href;
+  } catch (error) {
+    const msg = `Failure linking`;
+    console.log(`${msg} - person id:`, personId, "to reference link:", referenceLink);
+    console.log(msg, error);
+    throw new Error(msg);
+  }
+};
+
+const createReferenceLink = (patientNumber: number) => {
+  return `${apiUrl}/v1/org/${rootId}/patient/${patientNumber}${CW_ID_URL_ENCODED_PREFIX}${rootId}/`;
+};
+
+export const resetCommonwellLink = async (linkId: string) => {
+  try {
+    await commonWell.resetPatientLink(metriportQueryMeta, linkId);
+  } catch (error) {
+    const msg = `Failure resetting`;
+    console.log(`${msg} - link id:`, linkId);
+    console.log(msg, error);
+    throw new Error(msg);
+  }
+};
+
+export const getLinkFromCommonwell = async (
+  personId: string,
+  linkId: string
+): Promise<Link | void> => {
+  try {
+    // - get the patient link -> CommonWell.getPatientLink()
```

Comment:
nit: can update these comments to remove the `-`, extra spacing, and stuff like `-> CommonWell.getPatientLink()` since that's in the code now

Reply:
These comments was mostly just to show the logic being implemented ill remove them